rules/manager.go

```diff
@@ -743,6 +746,50 @@ func (g *Group) RestoreForState(ts time.Time) {
 
 }
 
+// Equals return if two groups have the same
+// 1. name
+// 2. file path
+// 3. interval
+// 4. rule set. and the rules are the same
+func (g *Group) Equals(ng *Group) bool {
+	if strings.Compare(g.name, ng.name) != 0 {
+		return false
+	}
+
+	if strings.Compare(g.file, ng.file) != 0 {
+		return false
+	}
+
+	if g.interval != ng.interval {
+		return false
+	}
+
+	if len(g.rules) != len(ng.rules) {
+		return false
+	}
+
+	gm := make(map[string]Rule, len(g.rules))
+	for _, r := range g.rules {
+		gm[r.Name()] = r
+	}
+
+	ngm := make(map[string]Rule, len(ng.rules))
+	for _, r := range ng.rules {
+		ngm[r.Name()] = r
+	}
```

Comment:
I am not sure if rule names can be repeated. Someone has to confirm.

Reply:
yes you can have multiple rules with same name

Reply:
fix it by using Rule.String() as a key